### DISCUSSION TITLE
0 instances predicted

### DISCUSSION THREAD

**Discussion by jramborger78** [talmolab/sleap#1600] [Help!]
Hi there! Have another little thing for you guys.

Basically, around ~300 Labels is when I got my best model, and since then have been trying to improve it with more labels but every since that one it barely predicts on any frames and has very poor recall. I am at about 700 labels right now and no matter the max stride, input scale, augmentations, etc, I can't seem to get it to make any predictions during training despite at times seeing the most loss I have seen yet using SLEAP, something that makes me excited that a model is improving.. Any ideas? Running a top-down model (as suggested by Talmo during a meeting with the George Lab) on a top-view video of a rat in an operant chamber. TThanks!

**Comment by talmo** [comment]
Hi @jramborger78,

Hmm, this is often symptomatic of bad/inconsistent labels which can sometimes slip into the dataset when you're working through multiple rounds of labeling.

Have you tried to go through and double check the existing labeled frames?

Could you also re-share the model training configuration from the **Run Training** window?

Thanks!

Talmo

**Reply by jramborger78** [reply]
Yeah happy to!I can poke trough my labels tomorrow when I can just run through them all. But I have attached a couple pictures for ya to possible elucidate some things.

First here is my skeleton given your comment on the poor instance labeling and consistency. Since this is a top down view and paws are often covered I tried to incorporate maybe the hips. More on that later.
![skeleton](https://github.com/talmolab/sleap/assets/134438857/14beca43-6f9b-4040-8c20-3952317b951e)

Second here is the best training/model results I got as mentioned in the original post (Was at 504 labels not 300, my apologies).
![training](https://github.com/talmolab/sleap/assets/134438857/f6ee5056-0548-430f-8b2c-7050281548d1)

Third here is a training after labeling some more predictions from the prior training in which I saw hardly any predictions made. This was the one I have kept as I'd delete the others to not muddy the folder but I tried I few augmentations to maybe fix the issue. But it is identical to the prior model. Wanted to maybe keep going simple before augmenting since We have many boxes with many varying lightings.
![train_554](https://github.com/talmolab/sleap/assets/134438857/ad99c23a-257b-44ae-b41a-82acbd545b2a)

And lastly here is the 700 label training where I noticed the worst images were of white rats so I labeled a ton in diverse poses and as you can see I started to adjust the max stride and filter rate after poking through other's questions here on github to you guys and it as a suggestion. I have tried many things to not end up with 0 predictions to no avail, including lowering learning rate and checking all augmentations, etc..
![train_705](https://github.com/talmolab/sleap/assets/134438857/a677c3a7-466e-4b47-8565-58d931687b08)

Since then I today have been relabeling a new model with a simpler skeleton. I removed the hips nodes as they were easily the least consistent ones to put down, especially on the white rats. The model does best on hoods and even brown rats, the edges by the back legs are easier to see to make "hips" for labels. Also felt the "spine" node, which is really its rear, was also difficult to be consistent. So in this newer one I changed that to "base" and it ONLY is placed at the base of the tail, and extended the "tail" node a little further down it, and also added a "catheter" node to be another ~centralized node in conjunction with the neck node to help the model map the body more than a jump from "neck" to some varying "spine" node.
